### PR TITLE
added pagination

### PR DESCRIPTION
**Pull request by Tushar09021999**
added pagination and sorting

### CODE REVIEW

File: src/main/java/com/crmapplication/crmapplication/controller/LeadController.java [hunk 1]
```diff
@@ -22,6 +24,17 @@ public class LeadController {
...
 
 
+
+
+    @GetMapping("/filtered-paginated")
+    public ResponseEntity<Page<Lead>> getLeadsByleadSourcePaginated(@RequestParam(name = "status") String leadSource,
+                                                                @RequestParam(defaultValue = "0") int page,
+                                                                @RequestParam(defaultValue = "10") int size) {
+        Page<Lead> leads = leadService.getLeadsByStatusPaginated(leadSource, PageRequest.of(page, size));
+        return ResponseEntity.ok(leads);
```
Comment (aMaZeDrago) [inline src/main/java/com/crmapplication/crmapplication/controller/LeadController.java:34]: add  logs. "getting lead by leadSource = leadSourceName"

File: src/main/java/com/crmapplication/crmapplication/controller/LeadController.java [hunk 2]
```diff
@@ -22,6 +24,17 @@ public class LeadController {
     private LeadServiceImpl leadService;
 
 
+
+
+    @GetMapping("/filtered-paginated")
+    public ResponseEntity<Page<Lead>> getLeadsByleadSourcePaginated(@RequestParam(name = "status") String leadSource,
+                                                                @RequestParam(defaultValue = "0") int page,
+                                                                @RequestParam(defaultValue = "10") int size) {
+        Page<Lead> leads = leadService.getLeadsByStatusPaginated(leadSource, PageRequest.of(page, size));
```
Comment (aMaZeDrago) [inline src/main/java/com/crmapplication/crmapplication/controller/LeadController.java:35]: add logs. "getting lead by leadSource = leadSourceName"

Comment (Tushar09021999) [reply src/main/java/com/crmapplication/crmapplication/controller/LeadController.java:35]: added

**Review by aMaZeDrago — CHANGES_REQUESTED** [review]
added logs

**Review by Tushar09021999 — COMMENTED** [review]
added

**Review by aMaZeDrago — APPROVED** [review]
Approved

**Pull request merged**